Modules/Loadable/Segmentations/SubjectHierarchyPlugins/qSlicerSubjectHierarchySegmentsPlugin.cxx

```diff
@@ -988,3 +1036,45 @@ bool qSlicerSubjectHierarchySegmentsPlugin::showItemInView(
     qSlicerSubjectHierarchyPluginHandler::instance()->pluginByName("Segmentations") );
   return segmentationsPlugin->showItemInView(segmentationItemId, viewNode, allItemsToShow);
 }
+
+//---------------------------------------------------------------------------
+void qSlicerSubjectHierarchySegmentsPlugin::setOpacityForCurrentItem(double opacity)
+{
+  // Get currently selected node and scene
+  vtkMRMLSubjectHierarchyNode* shNode = qSlicerSubjectHierarchyPluginHandler::instance()->subjectHierarchyNode();
```

Comment:
This 30 lines to get the display node and segment ID is very similar to those in `qSlicerSubjectHierarchySegmentsPlugin::showVisibilityContextMenuActionsForItem(vtkIdType itemID)`. It would be nice to factor it out into a helper function to avoid repetition.

Reply:
I thought about this, but the purpose of such convenience function is not clear. It could return the segmentation node, its display node, and the segment ID based on the item ID, but then the method signature would be quite confusing (we are not in Python where we can return tuples), and using it would not save a noticeable amount of lines of code, I'd guess about ten, or maybe nothing, depending on how we check the result.

Reply:
I had a look at this and the saving would be significant (about 80-100 lines of code) but the code would also become a bit more complex, so the overal gain is not much. The change would be also somewhat complicated and error-prone to do. Maybe this refactoring could be done later when other changes are made to this class.

Reply:
Thanks for looking into this! If we only consider the functional code I agree this could save several dozens of lines of code. As I was considering the new lines we would need to add (sanity checks, validating output, etc), then it started to seem that from the line count point of view it may not be worth it.